types/indexes.d.ts

```diff
@@ -93,6 +93,6 @@ declare module 'mongoose' {
      * ```
      */
     expires?: number | string;
-    weights?: AnyObject;
+    weights?: Record<string | number, number>;
```

Comment:
i am unsure if `string | number` is actually supported, should this maybe be changed to just `string`?

Reply:
I would say string only.